### PR TITLE
net-snmp update to latest recommended release (5.9.1)

### PR DESCRIPTION
**Pull request by glenn-horton-smith**
This adds the recently released version 5.9.1.  The net-snmp/CHANGES file says there are "Many bug fixes".

### CODE REVIEW

**Comment by glenn-horton-smith** [comment]
@darmac I see you created the net-snmp package. Would you like to be a maintainer with me?

File: var/spack/repos/builtin/packages/net-snmp/package.py [hunk 1]
```diff
@@ -10,9 +10,10 @@ class NetSnmp(AutotoolsPackage):
     """A SNMP application library, tools and daemon."""
 
     homepage = "http://www.net-snmp.org/"
-    url      = "https://sourceforge.net/projects/net-snmp/files/net-snmp/5.9/net-snmp-5.9.tar.gz"
+    url      = "https://sourceforge.net/projects/net-snmp/files/net-snmp/5.9.1/net-snmp-5.9.1.tar.gz/download"
 
     version('5.9', sha256='04303a66f85d6d8b16d3cc53bde50428877c82ab524e17591dfceaeb94df6071')
+    version('5.9.1', sha256='eb7fd4a44de6cddbffd9a92a85ad1309e5c1054fb9d5a7dd93079c8953f48c3f')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/net-snmp/package.py:15]: We usually put newer versions at the top, can you fix that?

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a minor request

File: var/spack/repos/builtin/packages/net-snmp/package.py (hunk 1 above)
Comment (glenn-horton-smith) [reply var/spack/repos/builtin/packages/net-snmp/package.py:15]: Thanks! Done.

**Review by glenn-horton-smith — COMMENTED** [review]
I noticed the build stage worked with jobs in parallel as is the default, but the "install" stage would not always work in parallel, complaining of directories not of found. The install stage works fine if done as only one job. I added code similar to that found in libelf. Please let me know if there is a better or more efficient way of doing this.

File: var/spack/repos/builtin/packages/net-snmp/package.py [hunk 2]
```diff
@@ -20,3 +21,6 @@ class NetSnmp(AutotoolsPackage):
     def configure_args(self):
         args = ['--with-defaults', 'LIBS=-ltinfo']
         return args
+
+    def install(self, spec, prefix):
+        make('install', parallel=False)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/net-snmp/package.py:26]: Read your comment. This is perfect to have only `make install` done serially.

**Review by alalazo — APPROVED** [review]

File: var/spack/repos/builtin/packages/net-snmp/package.py (hunk 2 above)
Comment (glenn-horton-smith) [reply var/spack/repos/builtin/packages/net-snmp/package.py:26]: Great, thanks for the feedback!

**Pull request merged**